### ISSUE TITLE
Bug in multiatomspace managing alpha equivalent links

### ISSUE THREAD

**Issue by ngeiswei** [opencog/atomspace#1226]
See unit test https://github.com/opencog/atomspace/blob/master/tests/atomspace/MultiSpaceUTest.cxxtest#L470

@linas I think the right thing to do is to have the alpha equivalent scope in the parent atomspace being returned when adding the scope in the child atomspace (which is what seems to happen for non scope atoms). Please correct me if I've got the wrong idea. Thanks.

**Comment by ngeiswei** [comment]
The problem comes from `AtomTable::getLinkHandle` that fails to identify that the scope is equivalent to the one in the parent atomspace. `getLinkHandle` recurses till it stumble on `Y` which is not in the parent atomspace, thus failing to fetch the whole scope.

I'm not sure yet what is the right way to fix that but I'm pretty sure that's the problem.

**Comment by linas** [comment]
The child atomspace is supposed to be as it has all of the atoms of the parent atomspace, and also some more.  This mostly makes sense, but there are some pathological cases, that seem hard to avoid, and I don't know how to solve them.

Say you add (Variable $X) to the child space, and then later on, add the same one to the parent.  The one in the child now hides the one in the parent, leading possibly to unexpected behavior ... and this is what your test case does:
```
		Handle as_A = as.add_node(CONCEPT_NODE, "A");
		Handle as_X = ex.add_node(VARIABLE_NODE, "X");
```
above creates (Variable X) in the child. The below creates a (variable X) in the parent, that is not the same as the one in the child: you now have two different atoms with the same type and name.  After that, its just confusing...
```
Handle as_XA = as.add_link(MEMBER_LINK, as_X, as_A);
```

**Comment by linas** [comment]
anyway, its not clear to me if the above is a unit-test bug, or if you meant to do that.

**Comment by ngeiswei** [comment]
I meant
```
Handle as_X = as.add_node(VARIABLE_NODE, "X");
```
this has been corrected in the fix proposal #1230
